tables/waf_traffic_log/waf_traffic_log.go

```diff
@@ -0,0 +1,130 @@
+	Location         *string  `json:"location"`
+	MatchedData      []string `json:"matchedData"`
+}
+
+type CustomValue struct {
+	Key   *string `json:"key"`
+	Name  *string `json:"name"`
+	Value *string `json:"value"`
+}
+
+// RateBasedRule represents the main JSON structure
+type RateBasedRule struct {
+	RateBasedRuleID     *interface{}  `json:"rateBasedRuleId"` // Assuming ID could be a string or number
+	RateBasedRuleName   *string       `json:"rateBasedRuleName"`
+	LimitKey            *string       `json:"limitKey"`
+	MaxRateAllowed      *int          `json:"maxRateAllowed"`
+	EvaluationWindowSec *int32        `json:"evaluationWindowSec"`
+	CustomValues        []CustomValue `json:"customValues" parquet:"type=JSON"`
+}
+
+type RuleMatchDetail struct {
+	ConditionType    *string  `json:"conditionType"`
+	SensitivityLevel *string  `json:"sensitivityLevel"`
+	Location         *string  `json:"location"`
+	MatchedData      []string `json:"matchedData"`
+}
+
+// Rule represents a rule entry
+type Rule struct {
+	RuleID           *string           `json:"ruleId"`
+	Action           *string           `json:"action"`
+	RuleMatchDetails []RuleMatchDetail `json:"ruleMatchDetails,omitempty"`
+	CaptchaResponse  CaptchaResponse   `json:"captchaResponse,omitempty"`
+}
+
+type CaptchaResponse struct {
+	ResponseCode   *int    `json:"responseCode,omitempty"`
+	SolveTimestamp *int64  `json:"solveTimestamp,omitempty"`
+	FailureReason  *string `json:"failureReason,omitempty"`
+}
+
+// RuleGroup represents the main JSON structure
+type RuleGroup struct {
+	RuleGroupID                 string `json:"ruleGroupId"`
+	TerminatingRule             *Rule  `json:"terminatingRule,omitempty"` // Can be null
+	NonTerminatingMatchingRules []Rule `json:"nonTerminatingMatchingRules"`
+	ExcludedRules               []Rule `json:"excludedRules,omitempty"` // Can be null
+}
+
+type Labels struct {
+	Name *string `json:"labels,omitempty"`
+}
+
+// WafTrafficLog struct with fields aligned to the provided JSON
+type WafTrafficLog struct {
+	schema.CommonFields
+
+	Action                      *string                `json:"action"`
+	CaptchaResponse             *CaptchaResponse       `json:"captchaResponse,omitempty" parquet:"name=captcha_response"`
+	FormatVersion               *int32                 `json:"formatVersion" parquet:"name=format_version"`
+	HttpRequest                 *HttpRequest           `json:"httpRequest,omitempty" parquet:"name=http_request"`
+	HttpSourceId                *string                `json:"httpSourceId,omitempty" parquet:"name=http_source_id"`
+	HttpSourceName              *string                `json:"httpSourceName,omitempty" parquet:"name=http_source_name"`
+	Labels                      []Labels               `json:"labels,omitempty" parquet:"type=JSON"`
+	NonTerminatingMatchingRules []Rule                 `json:"nonTerminatingMatchingRules,omitempty" parquet:"name=non_terminating_matching_rules, type=JSON"`
```

Comment:
```suggestion
	NonTerminatingMatchingRules []NonTerminatingRuleMatch                 `json:"nonTerminatingMatchingRules,omitempty" parquet:"name=non_terminating_matching_rules, type=JSON"`
```

To be consistent with the `TerminatingRuleMatchDetails` field, should this one also be updated?